documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+
+## Token Based Authorization with Keycloak Authorization Services
+
+Once the Kafka Broker has obtained an access token by using Strimzi Kafka OAuth for authentication, it is possible to use centrally managed authorization rules to enforce access restrictions onto Kafka Clients.
+For this Strimzi Kafka OAuth supports the use of `Keycloak Authorization Services`.
+
+A custom authorizer has to be configured on the Kafka Broker to take advantage of Authorization Services REST endpoints available on Keycloak, which provide a list of granted permissions on resources for authenticated users.
+The list of grants (permissions) is first fetched during the first action after authenticated session is established by Kafka client, and then regularly refreshed in the background.
+Grants are cached, and enforced locally on the Kafka Broker for each user session in order to provide fast authorization decisions. Because they are refreshed, any changes at Keycloak side are detected and enforced.
+
+
+## Starting Up the Pods
+
+NOTE: TODO: Change the url to: https://github.com/strimzi/strimzi-kafka-oauth/tree/master/examples/kubernetes
+
+For this example we're using the kubernetes deployment scripts available at https://github.com/mstruk/strimzi-kafka-oauth/tree/examples/examples/kubernetes
+
+NOTE: TODO: Change the url to: https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/master
+
+    export ROOT=https://raw.githubusercontent.com/mstruk/strimzi-kafka-oauth/examples
+
+.Deploy the Postgres database for the Keycloak
+
+    kubectl apply -f $ROOT/examples/kubernetes/postgres-pvc.yaml
+    kubectl apply -f $ROOT/examples/kubernetes/postgres.yaml
+
+.Deploy the Keycloak server
+
+    kubectl apply -f $ROOT/examples/kubernetes/keycloak-realms-configmap.yaml
+    kubectl apply -f $ROOT/examples/kubernetes/keycloak-postgres.yaml
+
+If your default namespace is not `myproject` but something else (for example, `default`), then use the following to deploy Keycloak:
+
+    curl -s $ROOT/examples/kubernetes/keycloak-postgres.yaml | sed -e "s#myproject#default#" | kubectl apply -f -
+
+.Deploy the minimal Kafka cluster
+
+Here we assume that the Strimzi Cluster Operator has already been installed on your Kubernetes cluster.
+
+    kubectl apply -f $ROOT/examples/kubernetes/kafka-oauth-single-authz.yaml
+
+
+## Using the Keycloak Admin Console to Configure Authorization
+
+You can login to the Keycloak Admin Console by creating a tunnel to the keycloak pod:
+
+    kubectl port-forward keycloak 8080
+
+Now use your browser to connect to `http://localhost:8080/auth/admin` and use `admin` as both a username, and a password.
+
+In the upper left corner under the Keycloak icon you should see `Master` selected as the current realm.
+Moving the mouse pointer over it should reveal two additional realms - `Demo` and `Kafka-authz`.
+
+For this example we are interested in the `kafka-authz` realm.
+Selecting it will open the `Realm Settings` for the `kafka-authz` realm.
```

Comment:
Could we drop the specifics from _In the upper left corner...to  ...for the `kafka-authz` realm._ and just say: _Locate the `kafka-authz` realm_?

Reply:
Highlight ui text with *bold*

Reply:
Absolutely.

Reply:
In fact, it's quite a bit more tricky. Let me think a bit about this.

Reply:
I hope I addressed it adequately.